File: src/main/java/com/dogGetDrunk/meetjyou/version/AppVersionRepository.java
package com.dogGetDrunk.meetjyou.version;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AppVersionRepository extends JpaRepository<AppVersion, Long> {
    Optional<AppVersion> findFirstByOrderByReleasedAtDesc();
    Optional<AppVersion> findByVersion(String version);
}

File: src/main/java/com/dogGetDrunk/meetjyou/version/AppVersionRepository.kt
package com.dogGetDrunk.meetjyou.version

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface AppVersionRepository : JpaRepository<AppVersion, Long> {
    fun findAllByPlatform(platform: Platform): List<AppVersion>
    fun findByVersionAndPlatform(version: String, platform: Platform): AppVersion?
}
